src/main/java/org/truffleruby/language/methods/Arity.java

```diff
@@ -90,6 +94,42 @@ public int getArityNumber() {
         return keywordArguments;
     }
 
+    public ArgumentDescriptor[] toAnonymousArgumentDescriptors() {
+        ArrayList<ArgumentDescriptor> descs = new ArrayList<>();
+
+        if (preRequired > 0) {
```

Comment:
Are these if statements necessary?

Reply:
They don't seem required indeed, so I think it would be clearer to remove them.

Reply:
I copied the logic from `Helpers.argsNodeToArgumentDescriptors`. I assumed it was done this way because we report negative values for arity as well. If that's not the case, I can remove the explicit checks.

Reply:
The loop condition already checks it so it's redundant. And that's legacy code :)

Reply:
Bah. I'm not sure what I was thinking.